### PR TITLE
Do not run packager in Release mode 

### PR DESCRIPTION
**Pull request by lucasbento**
## Summary

This PR fixes a regression introduced on [3273d23](https://github.com/facebook/react-native/commit/3273d23a2644e841877df2474cdec4a19f2253d2#diff-883359f85083d00b7266ec2acebcca9f) that removed #23938.

It also adds the check for `React` and `React-tvOS` targets.

## Changelog

[IOS] [FIXED] - Do not run packager in Release mode 

## Test Plan

Run RNTester in Release mode - packager shouldn't start anymore

### CODE REVIEW

**Comment by lucasbento** [comment]
@cpojer, @grabbou & @deepsweet: please review.

**Comment by cpojer** [comment]
Diff got stuck, let me try again.

**Comment by imben1109** [comment]
How can I check the fix version. Is this issue on 0.59 and fix in 0.60
